README.md

```diff
@@ -1,32 +1,239 @@
+
+## Installation
+
+**Note:** Tested and recommended installation on Ubuntu 22.04.1 LTS host OS
+
+### Docker Dependencies
+<details>
+<summary>Optional section for fresh install</summary>
+
+Uninstall existing docker
+```bash
+sudo apt-get remove docker docker-engine docker.io containerd runc
+```
+
+Install required packages
+```bash
+sudo apt-get update
+sudo apt-get install \
+    ca-certificates \
+    curl \
+    gnupg \
+    lsb-release
+```
 
+Add docker official GPG key
+```bash
+sudo mkdir -p /etc/apt/keyrings
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+```
 
-### Start Monitoring Server
-Run `docker compose up` in root folder.
+Setup docker repo
+```bash
+echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+  $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+```
 
+Update apt package index
+```bash
+sudo apt-get update
+```
+
+Install latest docker
+```bash
+sudo apt-get install docker-ce docker-ce-cli containerd.io docker-compose-plugin
+```
+</details>
 
-### Clients/Exporters Configuration
+### Nodies Monitoring
 
-- Promtail
+Git pulldown nodies monitoring project
+```bash
+git clone --branch main https://github.com/baaspoolsllc/nodies_monitoring.git
+```
 
+Install python3-pip
+```bash
+apt install python3-pip
+```
 
-In `clients/docker-compose.yml` under the promtail service configuration, edit `- ./log:/var/log` to the location of your nginx logs.
+Install requirements
+```bash
+pip3 install -r requirements.txt
+```
 
+## Setup
+
+Our project is split into a server(monitoring) stack, and a client(exporter) stack.
 
-In promtail/promtail-config.yml, edit clients.url to fit your monitoring server's hostname.
+### .env
+
+Create a .env file in the root [nodies_monitoring](./) folder
+
+This step is required on both the monitoring and the exporter stack.
+
+```
+SERVER_ENDPOINT=111.222.333.001
+CLIENT_ENDPOINT=111.222.333.002
+SLACK_WEBHOOK=https://hooks.slack.com/services/your_slack_webhook_string
 
-- BCExporter
+LOKI_PORT=3100
+PROMETHEUS_PORT=9090
+GRAFANA_PORT=3000
+MINIO_PORT=9000
+ALERTMANAGER_PORT=9093
 
+BLOCKCHAIN_EXPORTER_PORT=9877
+NODE_EXPORTER_PORT=9100
+CADVISOR_EXPORTER_PORT=8080
+PROMTAIL_PORT=9080
+```
 
-Add your chains.json to `clients/BCExporter/config/`. Check the readme for supported chains.
+- Update MONITORING_ENDPOINT with the ip address of the host that will run the [monitoring stack](./server) (loki, grafana, minio, prometheus, alertmanager)
 
+- Update EXPORTER_ENDPOINT with the ip address of the host that will run any services of the [exporter_stack](./clients) (blockchain_exporter, cadvisor, node_exporter, promtail)
 
-Configure your polling rate at `CExporter/config/config.yml`
+- Update SLACK_WEBHOOK with the webhook of the slack channel to send grafana-managed alerts to
 
-### Start clients
+### chains.json
```

Comment:
Can we include a sample `chains.json`, maybe something like `chains.example.json`